pkg/bpf/map_linux.go

```diff
@@ -88,6 +93,56 @@ type cacheEntry struct {
 	LastError     error
 }
 
+type historyEntry struct {
+	Timestamp time.Time
+	Key       MapKey
+	Value     MapValue
+	Action    DesiredAction
+	Err       error
+}
+
+type HistoryManager struct {
+	i       int
+	entries []*historyEntry
+}
+
+func (hm *HistoryManager) addEntry(key MapKey, value MapValue,
+	action DesiredAction, err error) {
+	entry := &historyEntry{
+		Timestamp: time.Now(),
+		Key:       key,
+		Value:     value,
+		Action:    action,
+		Err:       err,
+	}
+	// It max amount of entries is exceeded, get the subslice without the
+	// first element and then append the new entry.
+	if hm.i >= maxHistoryEntries {
+		hm.entries = hm.entries[1:maxHistoryEntries]
+		hm.entries = append(hm.entries, entry)
+		return
+	}
+	hm.i++
+	hm.entries = append(hm.entries, entry)
```

Comment:
Is there a protection against concurrent access of this slide?

Reply:
Sorry for that! I added the RWMutex.